### ISSUE TITLE
Unexpected training behaviour on Transformer model 

### ISSUE THREAD

**Issue by rubenaranamorera** [fastai/fastai#1897]
**Describe the bug**
I am not sure if something is wrong with Transformer model or I'm missing something, but I can not get improvements even on training loss on GAP dataset

**Provide your installation details**
```text
=== Software === 
python        : 3.6.5
fastai        : 1.0.50.post1
fastprogress  : 0.1.20
torch         : 1.0.1.post2
nvidia driver : 410.104
torch cuda    : 10.0.130 / is available
torch cudnn   : 7402 / is enabled

=== Hardware === 
nvidia gpus   : 1
torch devices : 1
  - gpu0      : 7618MB | Tesla M60

=== Environment === 
platform      : Linux-4.14.104-78.84.amzn1.x86_64-x86_64-with-glibc2.9
distro        : #1 SMP Mon Mar 4 19:19:37 UTC 2019
conda env     : Unknown
python        : /home/ec2-user/anaconda3/bin/python
sys.path      : 
/home/ec2-user/anaconda3/lib/python36.zip
/home/ec2-user/anaconda3/lib/python3.6
/home/ec2-user/anaconda3/lib/python3.6/lib-dynload
/home/ec2-user/anaconda3/lib/python3.6/site-packages
/home/ec2-user/anaconda3/lib/python3.6/site-packages/IPython/extensions
/home/ec2-user/.ipython
```

**To Reproduce**
````
from fastai.text import *
from tqdm import tqdm_notebook as tqdm


!wget https://github.com/google-research-datasets/gap-coreference/raw/master/gap-development.tsv -q
!wget https://github.com/google-research-datasets/gap-coreference/raw/master/gap-test.tsv -q
!wget https://github.com/google-research-datasets/gap-coreference/raw/master/gap-validation.tsv -q

data_path = Path(".")

train = pd.read_csv(data_path/"gap-development.tsv", sep="\t")
val = pd.read_csv(data_path/"gap-validation.tsv", sep="\t")
test = pd.read_csv(data_path/"gap-test.tsv", sep="\t")

train["is_valid"] = True
test["is_valid"] = False
val["is_valid"] = True
df_pretrain = pd.concat([train, test, val])

db = (TextList.from_df(df_pretrain, data_path/"db", cols="Text").split_from_df(col="is_valid").label_for_lm().databunch(bs=16))

awd_lstm_lm = language_model_learner(db, AWD_LSTM,  drop_mult=0.5, pretrained=True)
transformer_lm = language_model_learner(db, Transformer,  drop_mult=0.05, pretrained=True)

awd_lstm_lm.unfreeze()
transformer_lm.unfreeze()

awd_lstm_lm.fit_one_cycle(5, max_lr=1e-3)
transformer_lm.fit_one_cycle(5, max_lr=1e-3)
````

**Expected behavior**
I would expect the  to have similar results with Transformer and the AWD_LSTM, or at least to have some kind of nicer loss evolution on Transformer.

**Screenshots**

**Additional context**

I have changed drop_mult as suggested on the notebook:
https://github.com/cduguet/ulmfit-es/blob/master/ULMFit-TransfXL.ipynb

Which comes from a suggestion on a forum thread I can not access to even if I am logged in: 
https://forums.fast.ai/t/training-transformerxl/40104

In both cases, first there is awd_lstm_lm and then transformer:

![Captura de pantalla 2019-03-29 a las 13 24 03](https://user-images.githubusercontent.com/8973858/55232687-93000e00-5226-11e9-9a36-fe920f027402.png)

![Captura de pantalla 2019-03-29 a las 13 24 10](https://user-images.githubusercontent.com/8973858/55232695-96939500-5226-11e9-8fda-2ea644537c83.png)

**Comment by sgugger** [comment]
The pretrained transformer model is the first one from openAI, so it uses a different tokenization from fastai. This is probably why you have this bug. Let's continue the conversation on [the forum](https://forums.fast.ai/)
